### PR TITLE
[WIP]Example list should be better organized and more friendly

### PR DESCRIPTION
**Pull request by lloydsheng**
Per https://github.com/mapbox/ios-sdk-examples/issues/166
- [x]  Improve iPad user experience.  `UISplitViewController`
- [x]  Reorganize examples with categories 
- [ ]  China Localization

### CODE REVIEW

File: Examples/Examples.m [hunk 1]
```diff
@@ -10,63 +10,127 @@
...
+                    @{@"className": MBXExampleAnnotationView, @"title": @"Annotation views"},
+                    @{@"className": MBXExampleCalloutDelegateUsage, @"title": @"Callout delegate usage"},
+                    @{@"className": MBXExampleCustomCalloutView, @"title": @"Display custom views as callouts"},
+                    @{@"className": MBXExampleImageAnnotation, @"title": @"Mark a place on the map with an image"},
+                    @{@"className": MBXExampleWebAPIData, @"title": @"Dynamically style interactive points"},
+                    @{@"className": MBXExampleUserLocationAnnotation, @"title": @"Customize the user location annotation"},
+                    ]
+        },
+        @{
+            @"title": @"Markers and callouts",
```
Comment (frederoni) [inline Examples/Examples.m]: nit: duplicate of previous title, s/Markers and callouts/Annotations

**Comment by friedbunny** [comment]
@lloydsheng I missed your original post in the recent release bustle, but thanks for working on this.

As for the direction this PR should take, I’m not sure — any work should be done in coordination/consultation with @jmkiley and @captainbarbosa, who are the primary stewards of this repo and likely have thoughts about its future direction.

For instance:
- How to bring in title/description strings from `ios-sdk`?
- Is localizing this project something we want to maintain in the long run?

**Comment by lloydsheng** [comment]
> How to bring in title/description strings from ios-SDK?

We cloud sync titles/descriptions with https://www.mapbox.com/ios-sdk/examples/

> Is localizing this project something we want to maintain in the long run?

As far as i know, currently we should just maintain Chinese localization and China team can take over this work.

@friedbunny  Thanks for your suggestions, I'm not familiar with the workflow of `ios-sdk-examples`. @jmkiley @captainbarbosa please tell me your thoughts about this, I will cut up a ticket if we need further discussion.

**Comment by captainbarbosa** [comment]
@chriswu42 @1ec5 suggested maintaining a different branch on this repo for maintaining Chinese localization - a separate repository might be difficult to maintain. 

@lloydsheng this is a nice improvement! One issue I could see us running into is that not every example in this project is listed on our site. For example, the [`CameraFlyToExample.swift`](https://github.com/mapbox/ios-sdk-examples/blob/b330e07afa5b177f83ccb5aa2e05854a2c3e0589/Examples/Swift/CameraFlyToExample.swift) (and its Objective-C counterpart) isn't on our [examples page](https://www.mapbox.com/ios-sdk/examples/) but its in this project. When a new example is created here, we have to manually make a new page for it and assign it a title after the example for it has been created here so it can live on https://www.mapbox.com/ios-sdk/examples/.

**Comment by lloydsheng** [comment]
> This is a nice improvement! One issue I could see us running into is that not every example in this project is listed on our site.

@captainbarbosa  yes, here is iOS examples workflow i understood so far.
1. Implemented an example in `ios-SDK-examples`
2. Determine if we should write an awesome document about it (Which aspects we consider?)
3. If yes, the documentation should be in https://www.mapbox.com/ios-sdk/examples/

If my understanding is right, Number of examples inside `ios-sdk-examples` would more than documents.
I insist that we could think about the first two changes and merge them to master branch. because it's well organized and user-friendly in iPad iPhone. 🤓
About the China localization, Maintain it in an exclusive branch (cn-examples) is ok to me.

/cc @chriswu42

**Comment by chriswu42** [comment]
@lloydsheng are you planning to create this china branch?

@captainbarbosa anything we should be mindful of in maintaining this branch?

**Comment by captainbarbosa** [comment]
I'm 👍 with the proposed changes. 

In terms of keeping a `cn-examples` branch up-to-date, we regularly update the Podfile when new versions are released - this is part of the Maps SDK release checklist, but you'll have to pull in those changes when they merge to keep it up to date.

@lloydsheng in response to your question:

>Determine if we should prepare an awesome documentation presenting it (What are aspects we consider?)

Generally, when considering adding an example on the live page, proposed examples should:
- Should demonstrate one particular feature.
- Have minimal UIKit code so that buttons, switches, and sliders and the like don't distract from the example's goal.
-  Is not similar to other examples already illustrated on the page.

**Comment by lloydsheng** [comment]
>  are you planning to create this china branch?
@chriswu42  Yes, I've created  an `cn-examples` branch. Currently. It will support China localization and the work is in progress.

**Pull request merged**